docs/root/configuration/http_filters/router_filter.rst

```diff
@@ -136,6 +141,20 @@ Note that retry policies can also be applied at the :ref:`route level
 
 By default, Envoy will *not* perform retries unless you've configured them per above.
 
+.. _config_http_filters_router_x-envoy-retriable-status-codes:
+
+x-envoy-retriable-status-codes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Setting this header informs Envoy about what status codes should be considered retriable when used in
+conjunction with the :ref:`retriable-status-code <config_http_filters_router_x-envoy-retry-on>` retry policy.
+When the corresponding retry policy is set, the list of retriable status codes will be considered retriable
+in addition to the status codes enabled for retry through other retry policies. 
+
+The list is a comma delimited list of integers: "409" would cause 409 to be considered retriable, while "504,409"
+would consider both 504 and 409 retriable.
+
+This header will only be honored for requests from internal clients.
```

Comment:
Is there a cheap way to extend your tests to validate the internal-only behavior? Being a bit paranoid here as this could open up an amplification attack vector.

Reply:
I believe https://github.com/envoyproxy/envoy/pull/4597/files#diff-89ced54cfd0461279101d988ebdf45bbR489 covers it

Reply:
Do you send an external request and violate the header is ignored there? Maybe I'm missing it.